docs/testplan/transceiver_onboarding/cmis_cdb_firmware_upgrade_testplan.md

```diff
@@ -0,0 +1,489 @@
+
+4. A `cmis_cdb_firmware_base_url.csv` file (located in `ansible/files/transceiver_inventory` directory) should be present to define the base URL for downloading CMIS CDB firmware binaries. The file should follow this format:
+
+    ```csv
+    inv_name,fw_base_url
+    <inventory_file_name>,<base_url>
+    ```
+
+    - `inv_name`: The name of the inventory file that contains the definition of the target DUTs. For further details, please refer to the [Inventory File](https://github.com/sonic-net/sonic-mgmt/blob/master/docs/testbed/README.new.testbed.Configuration.md#inventory-file). The `inv_name` allows DUTs to be grouped based on their inventory file, enabling the test framework to fetch the correct base URL for firmware downloads.
+    - `fw_base_url`: The base URL from which the CMIS CDB firmware binaries can be downloaded. This URL should point to the directory where the firmware binaries are stored. e.g., `http://1.2.3.4/cmis_cdb_firmware/`.
+
+    Example of the file:
+
+    ```csv
+    inv_name,fw_base_url
+    lab,http://1.2.3.4/cmis_cdb_firmware/
+    ```
+
+5. A file (`sonic_{inv_name}_links.csv`) containing the connections of the ports should be present. This file is used to create the topology of the testbed which is required for minigraph generation.
+
+    - `inv_name` - inventory file name that contains the definition of the target DUTs. For further details, please refer to the [Inventory File](https://github.com/sonic-net/sonic-mgmt/blob/master/docs/testbed/README.new.testbed.Configuration.md#inventory-file)
+
+**Attributes for  the below tests**
+
+Following test attributes under `transceiver_firmware_upgrade` attribute category is applicable:
+
+| Attribute | Type | Default | Manadatory | Description |
+|-----------|------|---------|------------|-------------|
+| port_under_test | Dict | None | Yes | A dictionary containing the device name as the key and list of ports to be tested as its value |
+| firmware_versions | Dict | None | Yes | A dictionary containing the normalized transceiver product number as the key and list of firmware versions to be tested as its value |
+| firmware_download_timeout_minutes | Dict | 30 | No | A nested dictionary containing platform type as the outer key and normalized product number as the inner key and timeout value in minutes as the integer value |
+| restore_initial_firmwares | Bool | False | No | A flag indicating whether to restore the initial active and inactive firmware versions after testing is completed |
+| firmware_download_stress_iterations | Int | 5 | No | The number of iterations to stress test the firmware download process |
+| firmware_activation_stress_iterations | Int | 5 | No | The number of iterations to stress test the firmware activation process |
+| firmware_download_abort_method | String | "ctrl_c" | No | The method to abort the firmware download process. It can be one of the following strings : "ctrl_c", "sfputil_reset", "optic_reinsert" |
```

Comment:
@az-pz @mihirpat1 what about link up time timeout after new firmware activation?

Reply:
We can reuse `port_wait_time_after_startup_sec` attribute from `Link related tests` attribute category. Updated the TC#2 with the attribute.